extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketServerRecorder.java

```diff
@@ -254,6 +253,29 @@ public void handle(Void event) {
         };
     }
 
+    private static void logFailure(Throwable throwable, String message, WebSocketConnection connection) {
+        if (isWebSocketIsClosedFailure(throwable)) {
+            LOG.debugf(throwable,
+                    message + ": %s",
+                    connection);
+        } else {
+            LOG.errorf(throwable,
+                    message + ": %s",
+                    connection);
+        }
+    }
+
+    private static boolean isWebSocketIsClosedFailure(Throwable throwable) {
+        if (throwable == null) {
+            return false;
+        }
+        String message = throwable.getMessage();
+        if (message == null) {
+            return false;
+        }
+        return message.contains("WebSocket is closed");
```

Comment:
The exception is a `NoStackTraceException` so there is nothing to go by.

I do agree it's brittle, but I don't see any alternative and having useless ERROR log entries is worse IMHO.

Reply:
> The exception is a `NoStackTraceException` so there is nothing to go by.

Yes, this is something we could report in Vertx maybe? One thing we could do is to also test the connection with `WebSocketConnection.isClosed()`, i.e. `connections.isClosed() && message.contains("WebSocket is closed")` .

> I do agree it's brittle, but I don't see any alternative and having useless ERROR log entries is worse IMHO.

Well, the question is whether it's truly useless (some users might be interested in this) and as I pointed out the other question is how often does it actually happen?

We could also make this configurable, i.e. something like `quarkus.websockets-next.server.log-closed-connection-errors` with default value `false` :shrug:.

Reply:
> Well, the question is whether it's truly useless (some users might be interested in this) and as I pointed out the other question is how often does it actually happen?

I easily made it happen in tests. 

I also don't see why developers would need this in their logs, but let's see what @cescoffier thinks as well

Reply:
> It's a signal that something went wrong on the server 

Not when the client closed the connection before the server could send out the data

Reply:
> even say that it could be seen as a potential security issue if you can easily fill the logs with stacktraces by disconnecting clients

Definitely. 

Also, I am fairly sure we **don't** log these errors (or warnings) in our REST stack when the connection goes away before the response can be written

Reply:
> that we should avoid logging connection errors that can be easily initiated by the client

Like I mentioned above this should only happen infrequently - when the connection is closed during processing of a message or something like that. 

>  I would even say that it could be seen as a potential security issue if you can easily fill the logs with stacktraces by disconnecting clients.

I disagree. In that case, any server error would meet these conditions :thinking:.

But like I mentioned in https://github.com/quarkusio/quarkus/pull/40246#discussion_r1577894970 we should probably log a WARNING instead.

The only problem I can see is that the test relies on an exception message, which is super-brittle, may lead to false positives etc. We should at least combine `(exception instanceof VertxException) && connections.isClosed() && message.contains("WebSocket is closed")`

PS. I'm in the middle of a huge refactoring due to WebSocket client impl and I'm not going to merge any websocket-next PRs until the refactoring is finished :shrug:.

Reply:
> I disagree. In that case, any server error would meet these conditions 🤔.

The differentiator is when the user can trigger at will. That is why in Quarkus REST we don't log anything by default.

Reply:
> I disagree. In that case, any server error would meet these conditions

It actually really depends on the conditions.
See for instance: https://github.com/discourse/discourse/security/advisories/GHSA-89h3-g746-xmwq

> The only problem I can see is that the test relies on an exception message, which is super-brittle

That I agree. I would advise that we live with it (with your suggestion included if it works) until something in Vert.x allows us to do better (but we would need to create an issue there)?